### ISSUE TITLE
Bug with simpleitk_reader_writer.py when using nnUNetv2_find_best_configuration

### ISSUE THREAD

**Issue by arkiolok** [MIC-DKFZ/nnUNet#1496]
Hello,

I came across this error after training with the BraTS21 dataset and trying to use nnUNetv2_find_best_configuration.

```
Configuration 3d_lowres not found in plans nnUNetPlans.
Inferred plans file: /projects/hakanak2/nnUNet_prj/nnUNet_preprocessed/Dataset137_BraTS2021/nnUNetPlans.json.
Configuration 3d_cascade_fullres not found in plans nnUNetPlans.
Inferred plans file: /projects/hakanak2/nnUNet_prj/nnUNet_preprocessed/Dataset137_BraTS2021/nnUNetPlans.json.
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    ^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/nnUNet/nnunetv2/ensembling/ensemble.py", line 43, in merge_files
    image_reader_writer.write_seg(segmentation, output_filename_truncated + output_file_ending, properties)
  File "/home/hakanak2/nnUNet/nnunetv2/imageio/simpleitk_reader_writer.py", line 124, in write_seg
    itk_image = sitk.GetImageFromArray(seg.astype(np.uint8))
                                       ^^^^^^^^^^
AttributeError: 'Tensor' object has no attribute 'astype'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/hakanak2/miniconda3/envs/brainseg/bin/nnUNetv2_find_best_configuration", line 33, in <module>
    sys.exit(load_entry_point('nnunetv2', 'console_scripts', 'nnUNetv2_find_best_configuration')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/nnUNet/nnunetv2/evaluation/find_best_configuration.py", line 295, in find_best_configuration_entry_point
    find_best_configuration(dataset_name, model_dict, allow_ensembling=not args.disable_ensembling,
  File "/home/hakanak2/nnUNet/nnunetv2/evaluation/find_best_configuration.py", line 117, in find_best_configuration
    ensemble_crossvalidations([output_folder_1, output_folder_2], output_folder_ensemble, folds,
  File "/home/hakanak2/nnUNet/nnunetv2/ensembling/ensemble.py", line 193, in ensemble_crossvalidations
    _ = pool.starmap(
        ^^^^^^^^^^^^^
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 375, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 774, in get
    raise self._value
AttributeError: 'Tensor' object has no attribute 'astype'

```

Even though it was expected to be a ndarray in the function definition, seg comes out to be a Tensor, I've just changed `seg` with `seg.numpy()` in write_seg funciton on the line `itk_image = sitk.GetImageFromArray(seg.astype(np.uint8))
` to overcome this issue, and it did pass the ensembling phase, but after going for detecting the postprocessing, I've came up with the same error, but this time seg was an ndarray.

```
***Determining postprocessing for best model/ensemble***
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    ^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/nnUNet/nnunetv2/postprocessing/remove_connected_components.py", line 50, in load_postprocess_save
    image_reader_writer.write_seg(seg, output_fname, props)
  File "/home/hakanak2/nnUNet/nnunetv2/imageio/simpleitk_reader_writer.py", line 124, in write_seg
    itk_image = sitk.GetImageFromArray(seg.numpy().astype(np.uint8))
                                       ^^^^^^^^^
AttributeError: 'numpy.ndarray' object has no attribute 'numpy'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/hakanak2/miniconda3/envs/brainseg/bin/nnUNetv2_find_best_configuration", line 33, in <module>
    sys.exit(load_entry_point('nnunetv2', 'console_scripts', 'nnUNetv2_find_best_configuration')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/nnUNet/nnunetv2/evaluation/find_best_configuration.py", line 295, in find_best_configuration_entry_point
    find_best_configuration(dataset_name, model_dict, allow_ensembling=not args.disable_ensembling,
  File "/home/hakanak2/nnUNet/nnunetv2/evaluation/find_best_configuration.py", line 156, in find_best_configuration
    determine_postprocessing(all_results[best_key]['source'], join(nnUNet_raw, dataset_name, 'labelsTr'),
  File "/home/hakanak2/nnUNet/nnunetv2/postprocessing/remove_connected_components.py", line 128, in determine_postprocessing
    pool.starmap(
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 375, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hakanak2/miniconda3/envs/brainseg/lib/python3.11/multiprocessing/pool.py", line 774, in get
    raise self._value
AttributeError: 'numpy.ndarray' object has no attribute 'numpy'

```

Within the write_seg funciton, changing the `itk_image = sitk.GetImageFromArray(seg.astype(np.uint8))` line as:

```
if isinstance(seg, np.ndarray):
            itk_image = sitk.GetImageFromArray(seg.astype(np.uint8))
else:
            itk_image = sitk.GetImageFromArray(seg.numpy().astype(np.uint8))
```

let me pass this part. But I am not sure if this was the proper way. Wanted to inform you on that.

Best.

**Comment by michaelschwier** [comment]
I got the same issue. Any news on this regarding an official bugfix?

**Comment by izhorvath** [comment]
I found the same issue as well

**Comment by constantinulrich** [comment]
Hi, sorry for the delay. We just assigned multiple team members to work on all issues. 
This issue should have been solved two month ago by adding two lines [here](https://github.com/MIC-DKFZ/nnUNet/blob/5af388417c36b566ea029cc8b4e057d75a8ed422/nnunetv2/utilities/label_handling/label_handling.py#L181C1-L182C56). 
It is basically your fix @arkiolok  just at a different position.  
Let me know if it works for you.
